app/Views/Templates/components/forms/text-input.blade.php

```diff
@@ -0,0 +1,48 @@
+@props([
+    'inputType' => 'text',
+    'id' => '',
+    'name' => '',
+    'placeholder' => '',
+    'label' => '',
+    'size' => '',
+    'inputState' => '',
+    'value' => '',
+    'leadingVisual' => '',
+    'trailingVisual' => '',
+])
+
+@php
+    $sizeClass = $size ? 'input-'.$size : '';
+    $stateClass = $inputState ? 'input-'.$inputState : '';
+@endphp
+
+<div class='par relative w-full max-w-xs'>
+    @if($label)
+        <label for="{{ $id }}">
+            <span class="label-text">{{ $label }}</span>
+        </label>
+    @endif
+
+    <div class="relative">
+        @if($leadingVisual)
+            <span class="absolute inset-y-0 left-0 pl-3 flex items-center pointer-events-none">
+                <i class="{{ $leadingVisual }}"></i>
+            </span>
+        @endif
+
+        <input 
+            type="{{ $inputType }}" 
+            id="{{ $id }}" 
+            name="{{ $name }}" 
+            placeholder="{{ $placeholder }}" 
+            value="{{ $value }}" 
+            {{$attributes->merge(['class' => 'input input-bordered '.$sizeClass.' '.$stateClass.' '.'w-full max-w-xs '.($leadingVisual ? 'pl-10' : '').($trailingVisual ? 'pr-10' : '')])}}
```

Comment:
Since you are using the `$attribute` variable you don't actually need to define the other (standard) attributes (type, id, name, placeholder, value) They will all be part of the $attribute array and displayed alongside the class. (The merge function is really just a way to specifically merge 2 attributes as you do with the class).